plugins/mmdblookup/mmdblookup.c

```diff
@@ -429,58 +475,105 @@ CODESTARTdoAction
 	}
 
 
-	int status  = MMDB_get_entry_data_list(&result.entry, &entry_data_list);
+	MMDB_entry_data_s entry_data;
+	char value[1024];
+	int value_len = 0;
+	// Extract and amend fields (to message) as configured.
+	for (int i = 0; i < pData->fieldList.nmemb; ++i) {
+		size_t field_len = strnlen((char *)(pData->fieldList.name[i]), 1024);
+		char field[field_len + 1];
+		strncpy(field, (char *)pData->fieldList.name[i], sizeof(field));
+		field[field_len] = '\0'; // Ensure null termination.
+
+		mmdb_err = mmdb_get_value(&result.entry, &entry_data, field, field_len);
+		if (MMDB_SUCCESS != mmdb_err) {
+			dbgprintf("Got an error looking up the entry data - %s\n", MMDB_strerror(mmdb_err));
+			ABORT_FINALIZE(RS_RET_OK);
+		}
+		if (!entry_data.has_data || entry_data.offset <= 0) {
+			continue;
+		}
 
-	if (MMDB_SUCCESS != status) {
-		dbgprintf("Got an error looking up the entry data - %s\n", MMDB_strerror(status));
-		ABORT_FINALIZE(RS_RET_OK);
-	}
+		value_len = 0;
+		// Decode the value.
+		switch (entry_data.type) {
+		case MMDB_DATA_TYPE_BOOLEAN:
+			value_len = snprintf(value, sizeof(value), "%d", entry_data.boolean);
+			break;
+		case MMDB_DATA_TYPE_UTF8_STRING:
+			value_len = snprintf(
```

Comment:
I am not sure `snprintf` makes sense here. You could just copy the bytes directly. That said, could we avoid the copy entirely by using `json_object_new_string_len` here rather than copying to value and then `json_object_new_string` below?